main.go

```diff
@@ -21,21 +21,23 @@ import (
 	"k8s.io/klog/v2"
 
 	"k8s.io/kube-state-metrics/v2/internal"
+	"k8s.io/kube-state-metrics/v2/internal/discovery"
 	"k8s.io/kube-state-metrics/v2/pkg/options"
 )
 
 func main() {
 	opts := options.NewOptions()
 	cmd := options.InitCommand
 	cmd.Run = func(cmd *cobra.Command, args []string) {
-		internal.RunKubeStateMetricsWrapper(opts)
+		r := &discovery.GVKMaps{}
+		r.Client = discovery.NewDiscoveryClientFromOptions(opts)
+		go r.StartDiscovery()
```

Comment:
Would it make sense to move this logic into internal.RunKubeStateMetricsWrapper?